packages/nameguard-sdk/src/index.ts

```diff
@@ -365,34 +369,57 @@ export interface ConsolidatedNameGuardReport extends ConsolidatedReport {
   /* The ENSIP-15 normalization status of `name` */
   normalization: Normalization;
 
-  /** Beautified version of `name` */
+  /* Beautified version of `name` */
   beautiful_name: string;
+
+  /* Whether or not the name was inspected. */
+  inspected: boolean;
 }
 
 /**
  * NameGuard report that contains the full results of all `checks` on all `labels` in a name.
  */
-export interface NameGuardReport extends ConsolidatedNameGuardReport {
+export interface AbstractNameGuardReport extends ConsolidatedNameGuardReport {
   /* The results of all checks performed by NameGuard on `name`. */
   checks: CheckResult[];
 
-  /** Details of the inspection of all labels in `name`. */
-  labels: LabelGuardReport[];
+  /** Details of the inspection of all labels in `name`.
+    *
+    * `undefined` if `name` was uninspected and this is an `UninspectedNameGuardReport`.
+  */
+  labels?: LabelGuardReport[];
 
   /**
    * The name considered to be the canonical form of the analyzed `name`.
    *
-   * `null` if and only if the canonical form of `name` is considered to be undefined.
+   * `undefined` if and only if the canonical form of `name` is considered to be undefined or `name` was uninspected and this is an ``UninspectedNameGuardReport`.
    *
    * If a label is represented as `[labelhash]` in `name`,
    * the `canonical_name` will also contain the label represented as `[labelhash]`.
    *
    * `canonical_name` is guaranteed to be normalized with the exception of the case
    * where `normalization` is `unknown` and one or more labels are represented as `[labelhash]`.
    */
-  canonical_name: string | null;
+  canonical_name?: string;
+}
+
+export interface InspectedNameGuardReport extends AbstractNameGuardReport {
+    labels: LabelGuardReport[];
+    inspected: true;
 }
 
+/**
+ * NameGuard report for a name that was exceptionally long and was not inspected for performance reasons.
+ * Fields `label` and canonical_name` are undefined, `rating` is `alert`, `highest_risk` is `uninspected`, `risk_count` is `1`.
```

Comment:
I believe you can remove this entire line in the comment. Instead of writing these ideas in the comment, you can explicitly identify these fields in the interface definition here.

For example, just as you defined `inspected: false;` here, I believe you can also do the same for `rating`, `highest_risk`, and `risk_count`.